os/kernel/pthread/pthread_mutexunlock.c

```diff
@@ -83,6 +84,34 @@
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+/****************************************************************************
+ * Name: pthread_mutex_islocked
+ *
+ * Description:
+ *   Return true is the mutex is locked.
+ *
+ * Parameters:
+ *   None
+ *
+ * Return Value:
+ *   Returns true if the mutex is locked
+ *
+ ****************************************************************************/
+
+static inline bool pthread_mutex_islocked(FAR pthread_mutex_t *mutex)
+{
+	int semcount = mutex->sem.semcount;
```

Comment:
Kindly check whether mutex->sem != NULL or not before attempting to de-reference.
Thanks,
Pradeep